src/core/memory.d

```diff
@@ -148,7 +148,7 @@ private
     extern (C) void gc_removeRange(const void* p ) nothrow @nogc;
     extern (C) void gc_runFinalizers( const scope void[] segment );
 
-    package extern (C) bool gc_inFinalizer();
+    package extern (C) bool gc_inFinalizer() nothrow @nogc @safe;
```

Comment:
`core.gc.gcinterface.GC.inFinalizer` is only marked `nothrow`, and so is the extern(C) function. I could say we can probably set it to `@nogc`. `@safe` is probably fine too, but has a slight potential to break code (if there are any other private GC implementations out there).

But at the very least, the actual functions should be updated to reflect these attributes.

Reply:
Before I started working on this I thought that the only way to check if we're in a finalizer is to read the _`inFinalizer` static variable of the conservative GC implementation - I didn't expect my job to be so easy, as it was already exposed through the GC proxy, and so I forgot to update the proxy. Thanks for reminder, will fix.

On the actual attributes, I was not worried that they may put restrictions private GC implementation details, as after all, I think that in all implements this will amount to reading some variable, which should be safe in all circumstances (at worst the GC will need to add atomicLoads and stores, as it wouldn't be able to assume anything about the callers, but still, I don't see a big deal in that).